Automattic/jetpack
# Stats: Allow custom user role stats settings to be recognized

Fixes #8161

#### Changes proposed in this Pull Request:

* This PR adds custom roles to the SiteStatsComponent state. This means that the stats toggles for those custom roles are properly recognized when modified, and those changes also honored: custom user roles can view stats if that option is toggled on (and not otherwise), and similarly their logged in page views are recorded if that option is toggled on (and not otherwise).

#### Other information:


#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

Technically we're more reliably tracking non-standard user role activity or tracking how the settings are configured, but this is already [covered on our support doc](https://jetpack.com/support/wordpress-com-stats/#privacy).

#### Testing instructions:

* To test this out, you'll need to add several (more than one ideally) custom user roles on a Jetpack-connected site. The User Role Editor plugin is one way to do this.
* From a connected admin page, visit Jetpack > Settings > Traffic (`/wp-admin/admin.php?page=jetpack#/traffic`).
* In the 'Site Stats' section, test modifying the toggle state for various roles (custom and not custom). Do they correctly update? Try refreshing the page to make sure the changes have stuck.
* Then, logged in with the user roles in question, test to see if the changes applied are correctly reflected. 
  * For example, if the option to view stats is toggled on, can you view stats at this page, and is access denied if it is disabled? -> `/wp-admin/admin.php?page=stats`
  * Also, if the option to track logged in page views is enabled for any specific roles (custom and non custom), are logged in page views being correctly recorded (and not recorded if the toggle is disabled)?
